portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -58,4 +64,43 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     datastore.put(eventEntity);
     response.sendRedirect("/testForm.html");
   }
+  //TODO add doGet method
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Event");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+        for (Entity entity : results.asIterable()){
+            String organizer = (String) entity.getProperty("organizer");
+            String eventName = (String) entity.getProperty("eventName");
+            String location = (String) entity.getProperty("location");
+            String description = (String) entity.getProperty("description");
+            Date date = (Date) entity.getProperty("date");
+            EventData event = new EventData(organizer,eventName,location,description,date);
+            events.add(event);
+        }
+        String json_events = convertToJson(events);
+        response.setContentType("application/json;");
+        response.getWriter().println(json_events);
+  }
+   private String convertToJson(ArrayList items){
+      Gson gson = new Gson();
+      String json = gson.toJson(items);
+      return json;
+  }
+  private class EventData {
```

Comment:
Seems like you can use the Event class that already exists instead, and delete this private class: https://github.com/jamilao/sps-team-project/blob/master/portfolio/src/main/java/com/google/sps/servlets/Event.java